module/zfs/dsl_dataset.c

```diff
@@ -2353,6 +2353,10 @@ dsl_dataset_promote_check(void *arg, dmu_tx_t *tx)
 	}
 
 	snap = list_head(&ddpa->shared_snaps);
+	if (snap == NULL) {
+		err = SET_ERROR(EXDEV);
```

Comment:
@behlendorf 
thank you for suggest and explain for return code. I'm learn it.
I update it.
please review it again.